app/src/main/java/com/github/se/polyfit/di/UserModule.kt

```diff
@@ -102,4 +103,10 @@ object UserModule {
   fun providesFirebaseAuth(): FirebaseAuth {
     return FirebaseAuth.getInstance()
   }
+
+  @Provides
+  @Singleton
+  fun providesBarCodeCodeViewModel(): BarCodeCodeViewModel {
+    return BarCodeCodeViewModel()
+  }
```

Comment:
A potential issue I noticed is that because you added a `provides` function, only one instance of the ViewModel will be created. This means that if the user scans a few ingredients and proceeds to the recipe recommendation and then goes back to the home screen, or if after scanning they decide to go back, the next time they try to scan ingredients, the previous values will still be there. This happens because the list in the ViewModel is not cleared. This can be easily solved by deleting this `provides` function which will cause a new instance of the ViewModel to be created each time.